internal/getproviders/types.go

```diff
@@ -422,27 +422,24 @@ func VersionConstraintsString(spec VersionConstraints) string {
 			b.WriteString("??? ")
 		}
 
+		// The parser allows writing abbreviated versions (such as ~> 2) which
+		// end up being represented in memory with trailing unconstrained parts
```

Comment:
(dang it gh this is the 4th time I've written this comment PLEASE STICK THIS TIME)

I know this isn't a new comment, but I have a nitpicky request while you're in the area: reading this briefly (before my brain kicked in) lead me to expect that a constraint written as `~> 2.0` would be recorded as `2.0.0` (exact). It's silly, but can you clarify that comment? I don't know if my suggestion (below) is the right idea but something like it would have saved me a lot of time this morning 🙃

Reply:
I think your suggestion below is correct, but in rereading the comment I also spent 10 minutes confused, so I think it's probably structurally unclear. I'll take another shot at it. Thanks for pointing this out!

Reply:
Here's another attempt. Is this any clearer to you?

Reply:
YES thank you!! That's fantastic!